software/py/blood_graph_key.py

```diff
@@ -0,0 +1,48 @@
+import sys
+import argparse
+from blood_graph import BloodGraph
+from PIL import Image, ImageDraw, ImageFont
+from itertools import chain
+
+WIDTH  = 256
+HEIGHT = 256
+DEFAULT_MODE = "detailed"
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Depicts a visual key for bloodgraph")
+    parser.add_argument("--mode", default=DEFAULT_MODE, type=str,
+                        help="Type of bloodgraph - abstract / detailed")
+    args = parser.parse_args()
+
+    if (not (args.mode == "abstract" or args.mode == "detailed")):
+        parser.error("Invalid mode, can be one of abstract / detailed.")
+        sys.exit()
+
+    return args
+
+
+
+if __name__ == "__main__":
+    args = parse_args()
+
+    bg = BloodGraph(0,1,1, args.mode)
+    img = Image.new("RGB", (WIDTH, HEIGHT), "black")
+    draw = ImageDraw.Draw(img)
+    font = ImageFont.load_default()
+    # get the size of the font
+    yt = 0
+    # for each color in stalls...
+    for (key,color) in chain(bg.stall_bubble_color.iteritems(),
```

Comment:
Seems odd to name your iteration variable `key`, when you're drawing the key. Maybe cause would be better if you are iterating through the stall causes

Reply:
changed it to "operation".